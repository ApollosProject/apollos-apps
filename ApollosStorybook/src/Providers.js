import React from 'react';
import ApollosConfig from '@apollosproject/config';
import { Providers } from '@apollosproject/ui-kit';
import { MediaPlayerProvider } from '@apollosproject/ui-media-player';
import { LiveProvider } from '@apollosproject/ui-connected';
import ClientProvider, { client } from './client';
import customTheme, { customIcons } from './theme';

const AppProviders = (props) => (
  <ClientProvider>
    <MediaPlayerProvider>
        <LiveProvider>
          <Providers
            themeInput={customTheme}
            iconInput={customIcons}
            {...props}
          />
        </LiveProvider>
    </MediaPlayerProvider>
  </ClientProvider>
);

export default AppProviders;
