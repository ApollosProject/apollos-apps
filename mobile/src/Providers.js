import { nest } from 'recompose';

import ClientProvider from '/mobile/client';
import { ThemeProvider } from '/mobile/ui/theme';
import { WebBrowserProvider } from '/mobile/ui/WebBrowser';

export default nest(ClientProvider, ThemeProvider, WebBrowserProvider);
