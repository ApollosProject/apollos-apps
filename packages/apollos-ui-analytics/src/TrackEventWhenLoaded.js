import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import { AnalyticsConsumer } from './Provider';

class TrackEventWhenLoaded extends PureComponent {
  componentDidMount() {
    if (this.props.loaded || this.props.isLoading === false) {
      this.track();
    }
  }

  componentDidUpdate(prevProps) {
    if (
      (!prevProps.loaded && this.props.loaded) ||
      (prevProps.isLoading && !this.props.isLoading)
    ) {
      this.track();
    }
  }

  get trackClient() {
    return this.props.track;
  }

  track() {
    const { eventName, properties } = this.props;
    return this.trackClient({ eventName, properties });
  }

  render() {
    return null;
  }
}

TrackEventWhenLoaded.propTypes = {
  loaded: PropTypes.bool.isRequired,
  isLoading: PropTypes.bool.isRequired,
  eventName: PropTypes.string.isRequired,
  // eslint-disable-next-line react/forbid-prop-types
  properties: PropTypes.any,
  // eslint-disable-next-line react/forbid-prop-types
  track: PropTypes.func,
};

const TrackEventWhenLoadedConnected = (props) => (
  <AnalyticsConsumer>
    {({ track }) => <TrackEventWhenLoaded {...props} track={track} />}
  </AnalyticsConsumer>
);

export default TrackEventWhenLoadedConnected;
