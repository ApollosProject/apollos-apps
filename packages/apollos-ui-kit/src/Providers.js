import { nest } from 'recompose';

import { ThemeProvider } from './theme';
import { LayoutProvider } from './LayoutContext';

export default nest(ThemeProvider, LayoutProvider);
