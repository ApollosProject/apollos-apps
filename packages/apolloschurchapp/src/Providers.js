import React from 'react';
import { Providers } from '@apollosproject/ui-kit';
import { AuthProvider } from '@apollosproject/ui-auth';
import NavigationService from './NavigationService';
import { NotificationsManager } from './notifications';
import ClientProvider from './client';

const AppProviders = (props) => (
  <ClientProvider {...props}>
    <NotificationsManager>
      <AuthProvider navigateToAuth={() => NavigationService.navigate('Auth')}>
        <Providers {...props} />
      </AuthProvider>
    </NotificationsManager>
  </ClientProvider>
);

export default AppProviders;
