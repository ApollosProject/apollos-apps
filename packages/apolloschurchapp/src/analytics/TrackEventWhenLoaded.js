import React, { PureComponent } from 'react';
import { ApolloConsumer } from 'react-apollo';
import PropTypes from 'prop-types';
import { track } from './index';

class TrackEventWhenLoaded extends PureComponent {
  componentDidMount() {
    if (this.props.loaded) {
      this.track();
    }
  }

  componentDidUpdate(prevProps) {
    if (!prevProps.loaded && this.props.loaded) {
      this.track();
    }
  }

  get trackClient() {
    return this.props.track || track;
  }

  track() {
    const { eventName, properties, client } = this.props;
    return this.trackClient({ eventName, properties, client });
  }

  render() {
    return null;
  }
}

TrackEventWhenLoaded.propTypes = {
  loaded: PropTypes.bool.isRequired,
  eventName: PropTypes.string.isRequired,
  // eslint-disable-next-line react/forbid-prop-types
  properties: PropTypes.any,
  // eslint-disable-next-line react/forbid-prop-types
  client: PropTypes.any,
  track: PropTypes.func,
};

const TrackEventWhenLoadedConnected = (props) => (
  <ApolloConsumer>
    {(client) => <TrackEventWhenLoaded {...props} client={client} />}
  </ApolloConsumer>
);

export default TrackEventWhenLoadedConnected;
