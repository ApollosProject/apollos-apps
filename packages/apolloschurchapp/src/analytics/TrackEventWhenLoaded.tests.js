import React from 'react';
import renderer from 'react-test-renderer';

import Providers from 'apolloschurchapp/src/Providers';
import TrackEventWhenLoaded from './TrackEventWhenLoaded';

const trackMock = jest.fn();

describe('The TrackEventWhenLoaded component', () => {
  afterEach(() => {
    trackMock.mockClear();
  });
  it('should track an event if mounted with loaded=true', () => {
    renderer.create(
      <Providers>
        <TrackEventWhenLoaded
          loaded
          eventName="Track Something"
          properties={{ ThingId: '1234' }}
          track={trackMock}
        />
      </Providers>
    );
    expect(trackMock.mock.calls).toMatchSnapshot();
  });

  it('should track an event if mounted with loaded=false and then switching to true', () => {
    const tree = renderer.create(
      <Providers>
        <TrackEventWhenLoaded
          loaded={false}
          eventName="Track Something"
          properties={{ ThingId: '1234' }}
          track={trackMock}
          key="tracker"
        />
      </Providers>
    );
    expect(trackMock.mock.calls).toMatchSnapshot();
    tree.update(
      <Providers>
        <TrackEventWhenLoaded
          loaded
          eventName="Track Something"
          properties={{ ThingId: '1234' }}
          track={trackMock}
          key="tracker"
        />
      </Providers>
    );
    expect(trackMock.mock.calls).toMatchSnapshot();
  });
});
