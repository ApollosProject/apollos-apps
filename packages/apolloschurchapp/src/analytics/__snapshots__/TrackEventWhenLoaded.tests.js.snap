// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The TrackEventWhenLoaded component should track an event if mounted with loaded=false and then switching to true 1`] = `Array []`;

exports[`The TrackEventWhenLoaded component should track an event if mounted with loaded=false and then switching to true 2`] = `
Array [
  Array [
    Object {
      "client": ApolloClient {
        "__operations_cache__": Map {},
        "cache": InMemoryCache {
          "addTypename": true,
          "cacheKeyRoot": CacheKeyNode {
            "children": null,
            "key": null,
          },
          "config": Object {
            "addTypename": true,
            "dataIdFromObject": [Function],
            "fragmentMatcher": HeuristicFragmentMatcher {},
            "resultCaching": true,
          },
          "data": DepTrackingCache {
            "data": Object {},
            "depend": [Function],
          },
          "maybeBroadcastWatch": [Function],
          "optimisticData": DepTrackingCache {
            "data": Object {},
            "depend": [Function],
          },
          "silenceBroadcast": false,
          "storeReader": StoreReader {
            "cacheKeyRoot": CacheKeyNode {
              "children": null,
              "key": null,
            },
            "executeSelectionSet": [Function],
            "executeStoreQuery": [Function],
          },
          "storeWriter": StoreWriter {},
          "typenameDocumentCache": Map {},
          "watches": Set {},
        },
        "clearStoreCallbacks": Array [],
        "clientAwareness": Object {},
        "defaultOptions": Object {},
        "disableNetworkFetches": false,
        "link": ApolloLink {
          "request": [Function],
        },
        "localState": LocalState {
          "cache": InMemoryCache {
            "addTypename": true,
            "cacheKeyRoot": CacheKeyNode {
              "children": null,
              "key": null,
            },
            "config": Object {
              "addTypename": true,
              "dataIdFromObject": [Function],
              "fragmentMatcher": HeuristicFragmentMatcher {},
              "resultCaching": true,
            },
            "data": DepTrackingCache {
              "data": Object {},
              "depend": [Function],
            },
            "maybeBroadcastWatch": [Function],
            "optimisticData": DepTrackingCache {
              "data": Object {},
              "depend": [Function],
            },
            "silenceBroadcast": false,
            "storeReader": StoreReader {
              "cacheKeyRoot": CacheKeyNode {
                "children": null,
                "key": null,
              },
              "executeSelectionSet": [Function],
              "executeStoreQuery": [Function],
            },
            "storeWriter": StoreWriter {},
            "typenameDocumentCache": Map {},
            "watches": Set {},
          },
          "client": [Circular],
          "resolvers": Object {
            "Mutation": Object {
              "cacheMarkLoaded": [Function],
              "handleLogin": [Function],
              "logout": [Function],
              "mediaPlayerPlayNow": [Function],
              "mediaPlayerSetPlayhead": [Function],
              "mediaPlayerUpdateState": [Function],
              "updateDevicePushId": [Function],
              "updatePushPermissions": [Function],
            },
            "Query": Object {
              "authToken": [Function],
              "isLoggedIn": [Function],
              "notificationsEnabled": [Function],
            },
          },
        },
        "mutate": [Function],
        "query": [Function],
        "queryDeduplication": true,
        "reFetchObservableQueries": [Function],
        "resetStore": [Function],
        "resetStoreCallbacks": Array [],
        "ssrMode": false,
        "store": DataStore {
          "cache": InMemoryCache {
            "addTypename": true,
            "cacheKeyRoot": CacheKeyNode {
              "children": null,
              "key": null,
            },
            "config": Object {
              "addTypename": true,
              "dataIdFromObject": [Function],
              "fragmentMatcher": HeuristicFragmentMatcher {},
              "resultCaching": true,
            },
            "data": DepTrackingCache {
              "data": Object {},
              "depend": [Function],
            },
            "maybeBroadcastWatch": [Function],
            "optimisticData": DepTrackingCache {
              "data": Object {},
              "depend": [Function],
            },
            "silenceBroadcast": false,
            "storeReader": StoreReader {
              "cacheKeyRoot": CacheKeyNode {
                "children": null,
                "key": null,
              },
              "executeSelectionSet": [Function],
              "executeStoreQuery": [Function],
            },
            "storeWriter": StoreWriter {},
            "typenameDocumentCache": Map {},
            "watches": Set {},
          },
        },
        "typeDefs": undefined,
        "version": "2.5.1",
        "watchQuery": [Function],
      },
      "eventName": "Track Something",
      "properties": Object {
        "ThingId": "1234",
      },
    },
  ],
]
`;

exports[`The TrackEventWhenLoaded component should track an event if mounted with loaded=true 1`] = `
Array [
  Array [
    Object {
      "client": ApolloClient {
        "__operations_cache__": Map {},
        "cache": InMemoryCache {
          "addTypename": true,
          "cacheKeyRoot": CacheKeyNode {
            "children": null,
            "key": null,
          },
          "config": Object {
            "addTypename": true,
            "dataIdFromObject": [Function],
            "fragmentMatcher": HeuristicFragmentMatcher {},
            "resultCaching": true,
          },
          "data": DepTrackingCache {
            "data": Object {},
            "depend": [Function],
          },
          "maybeBroadcastWatch": [Function],
          "optimisticData": DepTrackingCache {
            "data": Object {},
            "depend": [Function],
          },
          "silenceBroadcast": false,
          "storeReader": StoreReader {
            "cacheKeyRoot": CacheKeyNode {
              "children": null,
              "key": null,
            },
            "executeSelectionSet": [Function],
            "executeStoreQuery": [Function],
          },
          "storeWriter": StoreWriter {},
          "typenameDocumentCache": Map {},
          "watches": Set {},
        },
        "clearStoreCallbacks": Array [],
        "clientAwareness": Object {},
        "defaultOptions": Object {},
        "disableNetworkFetches": false,
        "link": ApolloLink {
          "request": [Function],
        },
        "localState": LocalState {
          "cache": InMemoryCache {
            "addTypename": true,
            "cacheKeyRoot": CacheKeyNode {
              "children": null,
              "key": null,
            },
            "config": Object {
              "addTypename": true,
              "dataIdFromObject": [Function],
              "fragmentMatcher": HeuristicFragmentMatcher {},
              "resultCaching": true,
            },
            "data": DepTrackingCache {
              "data": Object {},
              "depend": [Function],
            },
            "maybeBroadcastWatch": [Function],
            "optimisticData": DepTrackingCache {
              "data": Object {},
              "depend": [Function],
            },
            "silenceBroadcast": false,
            "storeReader": StoreReader {
              "cacheKeyRoot": CacheKeyNode {
                "children": null,
                "key": null,
              },
              "executeSelectionSet": [Function],
              "executeStoreQuery": [Function],
            },
            "storeWriter": StoreWriter {},
            "typenameDocumentCache": Map {},
            "watches": Set {},
          },
          "client": [Circular],
          "resolvers": Object {
            "Mutation": Object {
              "cacheMarkLoaded": [Function],
              "handleLogin": [Function],
              "logout": [Function],
              "mediaPlayerPlayNow": [Function],
              "mediaPlayerSetPlayhead": [Function],
              "mediaPlayerUpdateState": [Function],
              "updateDevicePushId": [Function],
              "updatePushPermissions": [Function],
            },
            "Query": Object {
              "authToken": [Function],
              "isLoggedIn": [Function],
              "notificationsEnabled": [Function],
            },
          },
        },
        "mutate": [Function],
        "query": [Function],
        "queryDeduplication": true,
        "reFetchObservableQueries": [Function],
        "resetStore": [Function],
        "resetStoreCallbacks": Array [],
        "ssrMode": false,
        "store": DataStore {
          "cache": InMemoryCache {
            "addTypename": true,
            "cacheKeyRoot": CacheKeyNode {
              "children": null,
              "key": null,
            },
            "config": Object {
              "addTypename": true,
              "dataIdFromObject": [Function],
              "fragmentMatcher": HeuristicFragmentMatcher {},
              "resultCaching": true,
            },
            "data": DepTrackingCache {
              "data": Object {},
              "depend": [Function],
            },
            "maybeBroadcastWatch": [Function],
            "optimisticData": DepTrackingCache {
              "data": Object {},
              "depend": [Function],
            },
            "silenceBroadcast": false,
            "storeReader": StoreReader {
              "cacheKeyRoot": CacheKeyNode {
                "children": null,
                "key": null,
              },
              "executeSelectionSet": [Function],
              "executeStoreQuery": [Function],
            },
            "storeWriter": StoreWriter {},
            "typenameDocumentCache": Map {},
            "watches": Set {},
          },
        },
        "typeDefs": undefined,
        "version": "2.5.1",
        "watchQuery": [Function],
      },
      "eventName": "Track Something",
      "properties": Object {
        "ThingId": "1234",
      },
    },
  ],
]
`;
